fenixedu-ist-reports/src/main/java/pt/ist/fenix/jasper/reports/JasperReportPrinter.java

```diff
@@ -257,7 +260,7 @@ private JasperPrint createJasperPrint(final String key, final Map parameters, Co
                 dataSource = Collections.singletonList(StringUtils.EMPTY);
             }
 
-            return JasperFillManager.fillReport(report, parameters, new JRBeanCollectionDataSource(dataSource));
+            return JasperFillManager.fillReport(report, parameters, new TransactionalDataSource(dataSource));
```

Comment:
Why the new datasource type? The general practice of processing all the necessary information prior to the source construction is not applicable here? Seems safer than handling FF transaction inside the jasper engine.